setup/nextcloud-remove.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+#
+# This script will remove Nextcloud from your MiaB server
+##################################################################
+
+source setup/functions.sh # load our functions
+source /etc/mailinabox.conf # load global vars
+
+if [[ $EUID -ne 0 ]]; then
+	echo "This script must be run as root."
+	exit
+fi
+
+# Backup the existing ownCloud/Nextcloud.
+# Create a backup directory to store the current installation and database to
+
+BACKUP_DIRECTORY=$STORAGE_ROOT/owncloud-backup/`date +"%Y-%m-%d-%T"`
+mkdir -p "$BACKUP_DIRECTORY"
+if [ -d /usr/local/lib/owncloud/ ]; then
+	echo "Backing up existing Nextcloud installation, configuration, and database to directory to $BACKUP_DIRECTORY..."
+	cp -r /usr/local/lib/owncloud "$BACKUP_DIRECTORY/owncloud-install"
+	rm -r /usr/local/lib/owncloud
+fi
+if [ -e $STORAGE_ROOT/owncloud/owncloud.db ]; then
+	cp $STORAGE_ROOT/owncloud/owncloud.db $BACKUP_DIRECTORY
+fi
+if [ -e $STORAGE_ROOT/owncloud/config.php ]; then
+	cp $STORAGE_ROOT/owncloud/config.php $BACKUP_DIRECTORY
+fi
+if [ -d $STORAGE/owncloud/ ]; then
+	echo "Removing Nextcloud..."
+	rm -r $STORAGE_ROOT/owncloud
+fi
+# Remove Nextcloud's dependencies
+apt_purge php-imap php-pear php-dev php-xml php-zip php-apcu php-imagick
```

Comment:
Are you sure these dependencies are not required by anything else in MIAB?  How did you generate this list?

(Everything else looks fine in this script but I've not yet tested it.)

Reply:
Roundcube and Nextcloud have some dependencies in common.

I compared the dependencies of the two using the information from their scripts, and the ones that are needed by Nextcloud but not needed by Roundcube are removed.

Reply:
We don't need to support removing Nextcloud from a box that it's already been installed on. This can be commentary somewhere instead of a script (e.g. comments in the nextcloud setup script).